### PR TITLE
Add language, tz and peer_ip to mqtt context

### LINKED ISSUE #2742: Can't connect with MQTT client on local systems.
When you connect an MQTT client on localhost and use admin/admin as username password, zotonic crashes. This is because the peer_ip address is missing from the context when the username password is validated.

Because the peer ip is missing from the context, it can also lead to other problems. For instance with mod_ratelimit.

```erlang
08:51:16.362 [error] CRASH REPORT Process <0.12365.1> with 0 neighbours crashed with reason: {{badmap,undefined},[{erlang,map_get,[cowmachine_remote_ip,undefined],[]},{cowmachine_req,peer_ip,1,[{file,"/Users/mmzeeman/Work/git/zotonic/master/zotonic/_build/default/lib/cowmachine/src/cowmachine_req.erl"},{line,345}]},{m_identity,is_peer_allowed,1,[{file,"/Users/mmzeeman/Work/git/zotonic/master/zotonic/apps/zotonic_core/src/models/m_identity.erl"},{line,682}]},{m_identity,check_username_pw_1,3,[{file,"/Users/mmzeeman/Work/git/zotonic/master/zotonic/apps/zotonic_core/src/models/m_ident..."},...]},...]}
```

### PR DESCRIPTION
**Pull request by mmzeeman**
### Description

Fix #2742

Add language, timezone and peer_ip to the mqtt context.

### Checklist

- [ ] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

File: apps/zotonic_core/src/support/z_mqtt_sessions_runtime.erl [hunk 1]
```diff
@@ -97,13 +107,26 @@ maybe_set_language(#{ <<"preferences">> := #{ <<"language">> := Lang } }, Contex
         error:undef ->
             z_context:set_language(Lang, Context)
     end;
+maybe_set_language(#{ language := <<>> }, Context) ->
+    Context;
+maybe_set_language(#{ language := Lang }, Context) ->
+    try
```
Comment (mmzeeman) [inline apps/zotonic_core/src/support/z_mqtt_sessions_runtime.erl:113]: @mworrell Is this construct still needed? It looks like the function is always defined.

Comment (mworrell) [reply apps/zotonic_core/src/support/z_mqtt_sessions_runtime.erl:113]: I suspect it was a protection against illegal timezones.

Comment (mworrell) [reply apps/zotonic_core/src/support/z_mqtt_sessions_runtime.erl:113]: Are the language/timezone not always in the preferences?

Comment (mmzeeman) [reply apps/zotonic_core/src/support/z_mqtt_sessions_runtime.erl:113]: mod_translation could also be switched off. But in that case mod_translation is still callable. No `{error, undef}` is returned in that case.

**Pull request merged**